swap_meet/vendor.py

```diff
@@ -1,2 +1,110 @@
+from .item import Item
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self,inventory = None):
+        if inventory is None:
+            inventory = []
+        self.inventory = inventory
+
+    def add(self,item_to_added):
+        self.inventory.append(item_to_added)
+        return item_to_added
+    
+    def remove(self,item_to_removed):
+        if item_to_removed not in self.inventory:
+            return False
+
+        self.inventory.remove(item_to_removed)
+        return item_to_removed
+
+    def get_by_id(self, id):
+        #loop through the item in inventory
+        #if id input is equal to the item id in inventory then return item
+        for item in self.inventory:
+            if id == item.id:
+                return item
+        return None
+    
+    def swap_items(self, other_vendor, my_item, their_item):
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
```

Comment:
With longer compound conditionals such as this, keep in mind our PEP 8 requirements that lines be under 79 characters long! If you start to suspect you are getting close to that limit, feel free to break up the conditional statement across multiple lines. If you don't have parentheses or brackets to help Python intuit that multiple lines belong to the same statement, we can use the backslash (`\`) to indicate line continuation.

```suggestion
        if my_item not in self.inventory \
            or their_item not in other_vendor.inventory:
            return False
```